pyccel/parser/semantic.py

```diff
@@ -1010,7 +993,7 @@ def _create_variable(self, name, dtype, rhs, d_lhs, arr_in_multirets=False):
             is_temp = False
 
         if isinstance(rhs, (PythonTuple, InhomogeneousTupleVariable, NumpyNonZero)) or \
-                (isinstance(rhs, FunctionCall) and len(rhs.funcdef.results)>1):
+                ((isinstance(rhs, FunctionCall) and not isinstance(name, PyccelSymbol)) and len(rhs.funcdef.results)>1):
```

Comment:
What case does this additional check handle?

Reply:
Ah ok. So it is equivalent to:
```suggestion
                ((isinstance(rhs, FunctionCall) and rhs.pyccel_staging != 'syntactic') and len(rhs.funcdef.results)>1):
```
?
What is the type of `name` in this case?